modules/tensor_mechanics/include/userobjects/CrackFrontDefinition.h

```diff
@@ -162,6 +163,8 @@ class CrackFrontDefinition : public GeneralUserObject, public BoundaryRestrictab
   std::vector<bool> _is_point_on_intersecting_boundary;
   std::vector<Real> _j_integral_radius_inner;
   std::vector<Real> _j_integral_radius_outer;
+  const CrackFrontPointsProvider * _crack_front_points_provider;
+  unsigned int _number_crack_front_points;
```

Comment:
The name `_number_crack_front_points` could be confusing because it is specifically tied to the use of `crack_front_points_provider`.  Maybe something like `_num_points_from_provider` would be more clear.